crates/goose/src/providers/formats/openai.rs

```diff
@@ -253,6 +253,62 @@ pub fn get_usage(data: &Value) -> Result<Usage, ProviderError> {
     Ok(Usage::new(input_tokens, output_tokens, total_tokens))
 }
 
+/// Validates and fixes tool schemas to ensure they have proper parameter structure.
+/// If parameters exist, ensures they have properties and required fields, or removes parameters entirely.
+pub fn validate_tool_schemas(tools: &mut [Value]) {
+    for tool in tools.iter_mut() {
+        if let Some(function) = tool.get_mut("function") {
+            if let Some(parameters) = function.get_mut("parameters") {
+                if parameters.is_object() {
+                    ensure_valid_json_schema(parameters);
+                }
+            }
+        }
+    }
+}
+
+/// Ensures that the given JSON value follows the expected JSON Schema structure.
+fn ensure_valid_json_schema(schema: &mut Value) {
+    if let Some(params_obj) = schema.as_object_mut() {
+        // Check if this is meant to be an object type schema
+        let is_object_type = params_obj
+            .get("type")
+            .and_then(|t| t.as_str())
+            .map_or(true, |t| t == "object"); // Default to true if no type is specified
+
+        // Only apply full schema validation to object types
+        if is_object_type {
+            // Ensure properties field exists and is an object
+            if !params_obj.contains_key("properties") {
+                params_obj.insert("properties".to_string(), json!({}));
+            }
```

Comment:
nit: think you can simplify the logic here and the other 2 checks below with:
```suggestion
            params_obj.entry("properties").or_insert_with(|| json!({}));
```

https://docs.rs/serde_json/latest/serde_json/map/enum.Entry.html#method.or_insert_with

Reply:
good call